lib/widgets/content.dart

```diff
@@ -930,45 +917,30 @@ class _InlineContentBuilder {
   }
 
   InlineSpan _buildNode(InlineContentNode node) {
-    if (node is TextNode) {
-      return TextSpan(text: node.text, recognizer: _recognizer);
-    } else if (node is LineBreakInlineNode) {
+    return switch (node) {
+      TextNode() => TextSpan(text: node.text, recognizer: _recognizer),
       // Each `<br/>` is followed by a newline, which browsers apparently ignore
       // and our parser doesn't.  So don't do anything here.
-      return const TextSpan(text: "");
-    } else if (node is StrongNode) {
-      return _buildStrong(node);
-    } else if (node is DeletedNode) {
-      return _buildDeleted(node);
-    } else if (node is EmphasisNode) {
-      return _buildEmphasis(node);
-    } else if (node is LinkNode) {
-      return _buildLink(node);
-    } else if (node is InlineCodeNode) {
-      return _buildInlineCode(node);
-    } else if (node is UserMentionNode) {
-      return WidgetSpan(alignment: PlaceholderAlignment.middle,
-        child: UserMention(ambientTextStyle: widget.style, node: node));
-    } else if (node is UnicodeEmojiNode) {
-      return TextSpan(text: node.emojiUnicode, recognizer: _recognizer);
-    } else if (node is ImageEmojiNode) {
-      return WidgetSpan(alignment: PlaceholderAlignment.middle,
-        child: MessageImageEmoji(node: node));
-    } else if (node is MathInlineNode) {
-      return TextSpan(
+      LineBreakInlineNode() => const TextSpan(text: ""),
+      StrongNode() => _buildStrong(node),
+      DeletedNode() => _buildDeleted(node),
+      EmphasisNode() => _buildEmphasis(node),
+      LinkNode() => _buildLink(node),
+      InlineCodeNode() => _buildInlineCode(node),
+      UserMentionNode() => WidgetSpan(alignment: PlaceholderAlignment.middle,
+        child: UserMention(ambientTextStyle: widget.style, node: node)),
```

Comment:
Looking at these and thinking about the comment above, I think the arrangement I want is a switch statement with blank lines separating cases. Much like you did for `handleEvent` at my request above: https://github.com/zulip/zulip-flutter/pull/808#discussion_r1677246698

I'll just fix that up before merging. Then after that, I think the tiny helper methods here like `_buildEmphasis` make sense to inline, so I'll do that too.